app/src/main/java/com/davidtomas/taskyapp/features/auth/domain/useCase/ValidatePasswordUseCase.kt

```diff
@@ -1,19 +1,31 @@
 package com.davidtomas.taskyapp.features.auth.domain.useCase
 
-import com.davidtomas.taskyapp.features.auth.domain.model.LoginInputValidationError
+import com.davidtomas.taskyapp.features.auth.domain.model.InputValidationError
 import com.davidtomas.taskyapp.features.auth.domain.model.MissingMandatoryField
 import com.davidtomas.taskyapp.features.auth.domain.model.PasswordFormatError
+import com.davidtomas.taskyapp.features.auth.domain.model.PasswordLengthError
+import com.davidtomas.taskyapp.features.auth.domain.useCase.ValidatePasswordUseCase.ValidationType.REGISTER
 
 class ValidatePasswordUseCase {
-    operator fun invoke(password: String): LoginInputValidationError? =
-        when {
+    operator fun invoke(password: String, validationType: ValidationType): InputValidationError? {
+
+        val hasLowerCase = password.any { it.isLowerCase() }
+        val hasUpperCase = password.any { it.isUpperCase() }
+        val hasDigit = password.any { it.isDigit() }
+
+        return when {
             password.isEmpty() -> MissingMandatoryField
-            !password.matches(PASSWORD_REGEX.toRegex()) -> PasswordFormatError
+            password.length > MIN_PASSWORD_LENGTH && validationType == REGISTER -> PasswordLengthError
```

Comment:
Shouldn't this be `<`? You're returning an error if the password length is larger than the min password length